### PR TITLE
Align naming in Mode Select cluster XML with the spec.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
REVIEW NOTE: Only the first commit contains manual changes. The rest is code-generated.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@ limitations under the License.
 <configurator>
   <domain name="CHIP"/>
 
-  <struct name="SemanticTag">
+  <struct name="SemanticTagStruct">
     <cluster code="0x0050"/>
     <item name="MfgCode" type="ENUM16"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml:22]: My copy of the spec has ID0 for value and ID1 for MfgCode, which seems to imply this struct order needs to be reversed?

Or do we need to fix the spec if this was 1.0 released?

File: src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@ limitations under the License.
     <cluster code="0x0050"/>
     <item name="Label" type="CHAR_STRING" optional="false" length="32"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml:28]: My copy of the spec says max length here is 64

File: src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml:22]: Oh, nice catch...  Yes, this is broken and at this point that means it needs to be fixed in the spec.  Filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/6165

File: src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml:28]: This is ignored (for the moment), but good catch.  Will fix.

**Pull request merged**